### PR TITLE
Add CI for checking Input/Output/Attr of modified Ops 

### PR DESCRIPTION
**Pull request by liym27**
> Wiki about this CI: https://github.com/PaddlePaddle/Paddle/wiki/OP-Input-Output-Attribute-Compatibility-Modification

---
#### 1. Add check to compare op desc in PR_CI_CPU_Py2 
- This check to ensure that the trained model of the old version can be loaded with the new inference version.
- Please refer to https://github.com/PaddlePaddle/Paddle/pull/21530 for the interfaces used in this CI.

---
#### 2. The following are non conformances

- **for ops changed in branch PR:** 
  - add Input/Output but it's not dispensable
  - delele Input/Output
  - modify Input/Output including duplicable, intermediate or dispensable
  - add Attr but its default value is not set
  - delele Attr
  - modify Attr including the type, default value or  generated

- **for ops deleted in branch PR:** 
  - No comparison

- **for new ops only added in branch PR:** 
  - No comparison

---
#### 3.  Modify Op `fill_constant` and `slice`, the error message is as follows:

image

image

Further optimize error message：https://github.com/PaddlePaddle/Paddle/pull/21642

### CODE REVIEW

**Comment by luotao1** [comment]
> for ops deleted in branch PR: No comparison
   for new ops only added in branch PR: No comparison

@phlrain How do you think about it?

File: tools/check_api_approvals.sh [hunk 1]
```diff
@@ -73,6 +73,13 @@ if [ "$op_type_spec_diff" != "" ]; then
     check_approval 1 9301846 33742067 7913861
 fi
 
+# Todo(liym27): RD to be confirmed
+op_desc_diff=`python ${PADDLE_ROOT}/tools/check_op_desc.py ${PADDLE_ROOT}/paddle/fluid/OP_DESC_DEV.spec  ${PADDLE_ROOT}/paddle/OP_DESC_PR.spec`
+if [ "op_desc_diff" != "" ]; then
+    echo_line="You must have one RD (Aurelius84 or liym27 or zhhsplendid) approval for the changes of Inputs/Output/Attrs of Ops. The changes of Ops will cause that the new version inference fails to load model trained by the old version.\n"
```
Comment (luotao1) [inline tools/check_api_approvals.sh]: 此规范较为复杂，因此请加一个WIKI说明此CI的背景，正确的用法等。
可参考https://github.com/PaddlePaddle/Paddle/wiki/Operator-Development-Manual-Index 

Aurelius84 or liym27 or zhhsplendid，加上 @phlrain ，另需有一个Recommend

Comment (liym27) [reply tools/check_api_approvals.sh]: 感谢建议，已修改。相关wiki：https://github.com/PaddlePaddle/Paddle/wiki/OP-Input-Output-Attribute-Compatibility-Modification

**Review by luotao1 — APPROVED** [review]
LGTM！Excellent!

**Pull request merged**